File: python/2/2622.py
def main():
    number = int(input())

    result = 0
    for i in range(1, number):
        for j in range(1, i+1):
            k = number - i - j

            if not (k <= j):
                continue
            if not (i < j + k):
                continue

            result += 1

    print(result)


if __name__ == '__main__':
    main()

File: c/2622.c
#include <stdio.h>

int main() {
    int number;
    scanf("%d", &number);

    int k;
    int result = 0;
    for (int i = 1; i < number; i++) {
        for (int j = 1; j < i+1; j++) {
            k = number - i - j;

            if (!(k <= j))
                continue;
            if (!(i < j+k))
                continue;

            result++;
        }
    }

    printf("%d\n", result);

    return 0;
}
